parsing/parser.mly

```diff
@@ -472,6 +461,22 @@ let mk_quotedext ~loc (id, idloc, str, strloc, delim) =
   let e = ghexp ~loc (Pexp_constant (Pconst_string (str, strloc, delim))) in
   (exp_id, PStr [mkstrexp e []])
 
+let mk_indexop_expr array_indexing_operator ~loc
+      (array,dot,paren,index,set_expr) =
+  let assign = match set_expr with None -> false | Some _ -> true in
+  let n, index = array_indexing_operator.index loc paren index in
+  let fn = array_indexing_operator.name loc dot ~assign paren n in
+  let set_arg = match set_expr with
+    | None -> []
+    | Some expr -> [Nolabel, expr] in
+  let args = (Nolabel,array) :: index @ set_arg in
+  let attr =
+    { attr_name = mknoloc "__ocaml_index_op";
```

Comment:
This is not a reserved attribute: OCaml program might trigger the specialized error message by accident.

Reply:
So what would you recommend using instead? Would `"ocaml.index_op"` work?

I wanted to propose using a specific parsetree node (usually the nicer approach, even though it requires going through a painful amount of boilerplate modifications), but I am not sure how to do it in presence of user-defined indexing operators. Maybe actually make the identifier of the indexing function a parameter of the `Pexp_index_{get,set}` parsetree node?

Reply:
I don't think that user-indexed operators should be covered by the specialized error messages nor the parsetree node: contrarily to the array indexing operators they don't map to predetermined compiler primitives and they are not strangely altered by the `-unsafe` flag.
I believe that it is sensible to reflect this difference in the parsetree and error messages.